include/kernel.h

```diff
@@ -2427,7 +2427,7 @@ struct k_stack {
  * @req K-STACK-001
  */
 void k_stack_init(struct k_stack *stack,
-		  u32_t *buffer, unsigned int num_entries);
+		  u32_t *buffer, u32_t num_entries);
```

Comment:
Dueling pedantries here.

So... technically this is a public API change and not allowed without a deprecation stage for a release.  Obviously in practice it's not a problem.  But then obviously in practice this wasn't a problem to begin with.

Are we really putting MISRA compliance above rules that are designed to prevent code from breaking?  Meh.  I guess I don't really care, but...

Reply:
u32_t is a typecast for unsigned int. It doesn't really change the signature of the public API. So I am not sure if we have to go through with the whole deprecation cycle.

>Are we really putting MISRA compliance above rules that are designed to prevent code from breaking?

Misra is to prevent the code from breaking. I don't think this change will break for anyone.
Please provide an instance where this change will break.